agent/engine/data.go

```diff
@@ -0,0 +1,68 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package engine
+
+import (
+	apicontainer "github.com/aws/amazon-ecs-agent/agent/api/container"
+	apitask "github.com/aws/amazon-ecs-agent/agent/api/task"
+	"github.com/aws/amazon-ecs-agent/agent/data"
+	"github.com/aws/amazon-ecs-agent/agent/utils"
+
+	"github.com/cihub/seelog"
+)
+
+func (engine *DockerTaskEngine) saveTaskData(task *apitask.Task) {
+	err := engine.dataClient.SaveTask(task)
+	if err != nil {
+		seelog.Errorf("Failed to save data for task %s: %v", task.Arn, err)
```

Comment:
more of general question - wondering if failure to save should be critical? if save fails, do we just wait for the next time save is called?

Reply:
> if save fails, do we just wait for the next time save is called?

yes, this is same as the existing behavior. currently when saving with state manager, the error is also not treated as critical and is only logged https://github.com/aws/amazon-ecs-agent/blob/master/agent/engine/task_manager.go#L232-L236.
The only case we deal with save error is when first adding the task in acs handler https://github.com/aws/amazon-ecs-agent/blob/master/agent/acs/handler/payload_handler.go#L165-L171, in that case if we fail to save, we don't ack the payload (so the task will be sent again and we will try to save again). that behavior is maintained in this pr as well.